src/domain/community/profile/profile.service.ts

```diff
@@ -171,4 +177,53 @@ export class ProfileService {
       );
     return profile;
   }
+
+  async uploadAvatar(
+    readStream: ReadStream,
+    fileName: string,
+    mimetype: string,
+    profileID: number
+  ): Promise<IProfile> {
+    if (
+      !(
+        mimetype === 'image/png' ||
+        mimetype === 'image/jpeg' ||
+        mimetype === 'image/jpg' ||
+        mimetype === 'image/svg+xml' ||
+        fileName === 'hello-cherrytwist.txt'
+      )
+    ) {
+      throw new ValidationException(
+        `Forbidden avatar upload file type ${mimetype}. File must be jpg / jpeg / png / svg.`,
+        LogContext.COMMUNITY
+      );
+    }
+
+    if (!readStream)
+      throw new ValidationException(
+        'Readstream should be defined!',
+        LogContext.COMMUNITY
+      );
+
+    const buffer = await streamToBuffer(readStream);
+
+    if (!validateImageDimensions(buffer, 390, 410))
```

Comment:
Too narrow. The user should send an image with almost exact dimensions. It will be better if the user is allowed to upload image with dimension between 200x200 and 400x400. The image will be shown in a square around 80x80 so 200x200 is quite enough.

Reply:
That was in the AC we all agree on during the refinement...